openforcefield/tests/test_forcefield.py

```diff
@@ -1557,6 +1594,437 @@ def test_lookup_parameter_handler_object(self):
+
+    # Test the wildcard match
+    opts = OrderedDict(
+        {
+            "xml": xml_ff_virtual_sites_bondcharge_match_once,
+            "smi": "[C][C]",
+            "assert_physics": (
+                # charge        sigma         epsilon
+                (+0.1 * q_unit, None, None),
+                (+0.1 * q_unit, None, None),
+                (-0.2 * q_unit, 0.1 * s_unit, 0.1 * e_unit),
+            ),
+        }
+    )
+    bond_charge_parameters_args.append(opts)
+
+    # Test the wildcard match
+    opts = OrderedDict(
+        {
+            "xml": xml_ff_virtual_sites_bondcharge_match_all,
+            "smi": "[N]#[N]",
+            "assert_physics": (
+                # charge        sigma         epsilon
+                (+0.4 * q_unit, None, None),
+                (+0.4 * q_unit, None, None),
+                (-0.4 * q_unit, 0.2 * s_unit, 0.2 * e_unit),
+                (-0.4 * q_unit, 0.2 * s_unit, 0.2 * e_unit),
+            ),
+        }
+    )
+    bond_charge_parameters_args.append(opts)
+
+    ############################################################################
+    # Monovalent virtual site test data
+    ############################################################################
+
+    # Test that using different names allows for multiple virtual sites
+    opts = OrderedDict(
+        {
+            "xml": xml_ff_virtual_sites_bondcharge_match_once_two_names,
+            "smi": "[N]#[N]",
+            "assert_physics": (
+                # charge        sigma         epsilon
+                (+0.3 * q_unit, None, None),
+                (+0.3 * q_unit, None, None),
+                (-0.2 * q_unit, 0.1 * s_unit, 0.1 * e_unit),
+                (-0.4 * q_unit, 0.2 * s_unit, 0.2 * e_unit),
+            ),
+        }
+    )
+    bond_charge_parameters_args.append(opts)
+
+    opts = OrderedDict(
+        {
+            "xml": xml_ff_virtual_sites_monovalent_match_once,
+            "smi": None,
+            "assert_physics": (
+                # charge        sigma         epsilon
+                (+0.2 * q_unit, None, None),
+                (+0.2 * q_unit, None, None),
+                (+0.2 * q_unit, None, None),
+                (+0.0 * q_unit, None, None),
+                (+0.0 * q_unit, None, None),
+                (+0.0 * q_unit, None, None),
+                (+0.0 * q_unit, None, None),
+                (-0.6 * q_unit, 0.2 * s_unit, 0.2 * e_unit),
+            ),
+            "mol": create_acetaldehyde(),
+        }
+    )
+    monovalent_parameters_args.append(opts)
+
+    ############################################################################
+    # Divalent virtual site test data
+    ############################################################################
+
+    # A TIP5 definition from OpenMM
+    opts = OrderedDict(
+        {
+            "xml": xml_ff_virtual_sites_divalent_match_all,
+            "smi": None,
+            "assert_physics": (
```

Comment:
Might it also be useful to verify the mass here? Although it may not be worth it if you have to reach somewhere other than `NonbondedForce`